corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
```

Comment:
My thought process here was that I want this lookup to be done at most once per request.  I'm not quite certain if I should even worry about this, but I was thinking that the real caching is largely unnecessary, and the in-memory component would suffice, so I should skip over the real caching part of it.

Basically I saw the same exact db call being performed many times in one request, so rather than refactor the code to look that up once and pass the result around, I'd just memoize the lookup for the duration of the request.

Having said that, it looks like this might actually get called with the same parameters a lot more often than I'd thought, and WOULD benefit from a proper (skippable) cache.